tools/dkcm/clerk/clerk.go

```diff
@@ -75,77 +76,199 @@ type ClerkOperations interface {
 func NewDB(c *mysql.DBConfig) (*DB, error) {
-	return nil, nil
+	db, err := c.Connect()
+	return &DB{db}, err
 }
 
-func (db *DB) CheckAvail() (bool, string, string) {
-	return true, "", ""
+// check available clusters in Cluster db and return zone and projectID that would be used by ProwJob
+func (db *DB) CheckAvail(nodes int, nodeType string, zone string) (bool, int64) {
+	// check whether available cluster exists
+	row := db.QueryRow("SELECT * FROM Cluster WHERE Status = 'Ready' AND Nodes = ? AND NodeType = ? AND Zone = ?",
+		nodes, nodeType, zone)
+	c, err := populateCluster(row)
+	// no available cluster is found
+	if err != nil {
+		return false, 0
+	}
+	return true, c.ClusterID
+
 }
 
 // insert a cluster entry into db
-func (db *DB) InsertCluster(nodes int, nodeType string, accessToken string, projectID string, prowJobID string, status string, zone string) error {
-	return nil
+func (db *DB) InsertCluster(nodes int, nodeType string, projectID string, status string, zone string) error {
+	stmt, err := db.Prepare(`INSERT INTO Cluster(Nodes, NodeType, ProjectID, Status, Zone)
+							VALUES (?,?,?,?,?,?)`)
+	defer stmt.Close()
+
+	if err != nil {
+		return err
+	}
+
+	_, err = stmt.Exec(nodes, nodeType, projectID, status, zone)
+	return err
+}
+
+func updateQueryString(dbName string, id int, fields []string, values []interface{}) string {
+	updateStatement := fmt.Sprintf(`UPDATE %s SET `, dbName)
+	idStatement := fmt.Sprintf(` WHERE ID = % d`, id)
+	var fieldStatements []string
+	for idx, val := range fields {
+		fieldStatements = append(fieldStatements, fmt.Sprintf("%s = %v", val, values[idx]))
+	}
+	return updateStatement + strings.Join(fieldStatements, `, `) + idStatement
 }
 
 // Update a cluster entry on different fields len(fields) == len(values)
-func (db *DB) UpdateCluster(clusterID string, fields []string, values []string) error {
-	return nil
+func (db *DB) UpdateCluster(clusterID int, fields []string, values []interface{}) error {
+	queryString := updateQueryString("Cluster", clusterID, fields, values)
+	_, err := db.Exec(queryString)
+	return err
 }
 
-// List all clusters within a certain time window
-func (db *DB) ListClusters(window time.Duration) ([]Cluster, error) {
+// List all clusters
+func (db *DB) ListClusters() ([]Cluster, error) {
 	var result []Cluster
+	rows, err := db.Query(`
+	SELECT ID, Nodes, NodeType, ProjectID, Status, Zone
+	FROM Cluster`)
+	if err != nil {
+		return result, err
+	}
+	for rows.Next() {
+		c, err := populateCluster(rows)
+		if err != nil {
+			return result, err
+		}
+		result = append(result, *c)
+	}
 	return result, nil
 
 }
 
 // DeleteCluster deletes a row from Cluster db
-func (db *DB) DeleteCluster(accessToken string) error {
-	return nil
+func (db *DB) DeleteCluster(clusterID int) error {
+	delString := "DELETE FROM Cluster WHERE ID = ?"
+	stmt, err := db.Prepare(delString)
+	defer stmt.Close()
+	if err == nil {
+		err = checkAffected(stmt, 1, clusterID)
```

Comment:
If you worry about wrong number of rows deleted, might want to do `GetCluster` before delete and make sure the row to be deleted exist, so that the problem can be better isolated